sa/sa.go

```diff
@@ -162,6 +165,111 @@ func (ssa *SQLStorageAuthority) UpdateRegistration(ctx context.Context, req *cor
 	return &emptypb.Empty{}, nil
 }
 
+// UpdateRegistrationContact stores an updated contact in a Registration
+func (ssa *SQLStorageAuthority) UpdateRegistrationContact(ctx context.Context, req *sapb.UpdateRegistrationContactRequest) (*corepb.Registration, error) {
+	if req == nil || req.RegistrationID == 0 {
+		return nil, errIncompleteRequest
+	}
+
+	// We don't want to write literal JSON "null" strings into the database if the
+	// list of contact addresses is empty. Replace any possibly-`nil` slice with
+	// an empty JSON array.
+	jsonContact := []byte("[]")
+	var err error
+	if len(req.Contact) != 0 {
+		jsonContact, err = json.Marshal(req.Contact)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	_, overallError := db.WithTransaction(ctx, ssa.dbMap, func(tx db.Executor) (interface{}, error) {
+		result, err := tx.ExecContext(ctx,
+			"UPDATE registrations SET contact = ? WHERE id = ? LIMIT 1",
+			jsonContact,
+			req.RegistrationID,
+		)
+		if err != nil {
+			return nil, err
+		}
+		rowsAffected, err := result.RowsAffected()
+		if err != nil || rowsAffected != 1 {
+			return nil, berrors.InternalServerError("no registration ID '%d' updated with new contact field", req.RegistrationID)
+		}
+
+		updatedRegistration, err := selectRegistration(ctx, tx, "id", req.RegistrationID)
+		if err != nil {
+			if db.IsNoRows(err) {
+				return nil, berrors.NotFoundError("registration with ID '%d' not found", req.RegistrationID)
+			}
+			return nil, err
+		}
+
+		return updatedRegistration, nil
```

Comment:
This `updatedRegistration` is being thrown away on line 186, where you assign it to the empty variable `_`. See [NewOrderAndAuthzs](https://github.com/letsencrypt/boulder/blob/a731497958e57bada9a4eeb358edb5e45b519309/sa/sa.go#L474) for the right way to [recover output from a transaction](https://github.com/letsencrypt/boulder/blob/a731497958e57bada9a4eeb358edb5e45b519309/sa/sa.go#L614-L617) like this, and then you'll have to return that result on line 214.

Reply:
Thanks! In reworking this, I realized that calling the error value `overallError` for the transaction - as opposed to the usual `err` - was making it easy to mislead myself about the control flow. Is it idiomatic to use `overallError` in this context, or OK to stop?

Reply:
We do use `overallError` in this context in multiple places around the SA. You're right that it's somewhat non-idiomatic, but for now I'd prefer to match the pattern used in the other SA methods.